### PR TITLE
Wallet bridge

### PR DESCRIPTION
**Pull request by samsiegart**
refs #4 
refs #9
refs #11 

[Blocked] https://github.com/Agoric/dapp-inter/pull/28/commits/79241137a89c29d23d3e528ab52163f2aa2e6911 requires https://github.com/Agoric/agoric-sdk/pull/6757 to work

### CODE REVIEW

File: src/store/app.ts [hunk 1]
```diff
@@ -52,3 +106,25 @@ export const networkConfigAtom = atomWithStorage(
   'agoric-network-config',
   networkConfigs.mainnet,
 );
+
+const prodBridgeHref = 'https://wallet.agoric.app/wallet/bridge.html';
+const localBridgeHref = 'http://localhost:3000/wallet/bridge.html';
+const branchBridgeHref = (branchName: string) =>
+  `https://${branchName}.wallet-app.pages.dev/wallet/bridge.html`;
```
Comment (turadg) [inline src/store/app.ts:113]: consider a config or urls module

File: src/vite-env.d.ts [hunk 2]
```diff
@@ -10,3 +10,34 @@ declare module '@agoric/ui-components' {
 declare module '@agoric/wallet/api/src/marshal-contexts' {
   export const makeImportContext;
 }
+
+declare module '@agoric/web-components' {
```
Comment (turadg) [inline src/vite-env.d.ts:14]: until https://github.com/Agoric/agoric-sdk/issues/6591

File: src/components/OfferSignerBridge.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+import {
+  offerSignerAtom,
+  chainConnectionAtom,
+  bridgeHrefAtom,
+  walletUiHrefAtom,
+  setIsDappApprovedAtom,
+} from 'store/app';
+import type { OfferConfig } from 'store/app';
+
+type BridgeReadyMessage = {
```
[the remark this answers is not in the document]
Comment (samsiegart) [reply src/components/OfferSignerBridge.tsx:16]: Done

File: src/components/OfferSignerBridge.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+      { autoClose: 5000 },
+    );
+  };
+
+  const onBridgeReady = (ev: BridgeReadyMessage) => {
+    const {
+      detail: { isDappApproved, requestDappConnection, addOffer },
+    } = ev;
+    if (!isDappApproved) {
+      requestDappConnection('Inter Protocol PSM');
```
[the remark this answers is not in the document]
Comment (samsiegart) [reply src/components/OfferSignerBridge.tsx]: Oops copy-pasta, fixed

File: src/components/OfferSignerBridge.tsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+      </p>,
+      { autoClose: 5000 },
+    );
+  };
+
+  const onBridgeReady = (ev: BridgeReadyMessage) => {
+    const {
+      detail: { isDappApproved, requestDappConnection, addOffer },
+    } = ev;
+    if (!isDappApproved) {
```
[the remark this answers is not in the document]
Comment (samsiegart) [reply src/components/OfferSignerBridge.tsx:104]: Ah right, done.

File: src/components/OfferSignerBridge.tsx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+          </a>
+          {message && `: ${message}`}
+        </p>
+      </div>,
+    );
+  };
+
+  const onBridgeMessage = (ev: BridgeMessage) => {
+    const data = ev.detail.data;
+    const type = data.type;
```
[the remark this answers is not in the document]
Comment (samsiegart) [reply src/components/OfferSignerBridge.tsx]: Done

File: src/components/OfferSignerBridge.tsx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+    switch (type) {
```
[the remark this answers is not in the document]
Comment (samsiegart) [reply src/components/OfferSignerBridge.tsx]: Done

File: src/service/wallet.ts [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+import type { Id as ToastId, ToastContent, ToastOptions } from 'react-toastify';
+
+export const watchPurses = (chainConnection: { pursesNotifier: unknown }) => {
+  let isCancelled = false;
+
+  const watch = async () => {
+    const n = chainConnection.pursesNotifier;
+
+    for await (const purses of iterateNotifier(n)) {
+      if (isCancelled) return;
```
[the remark this answers is not in the document]
Comment (samsiegart) [reply src/service/wallet.ts:18]: If `walletService.disconnect()` gets called. Currently that doesn't happen anywhere, but it's something we wanted to add in the PSM but never did. Just added a TODO to the `ConnectWalletButton` so we can add a disconnect option or something.

File: src/service/wallet.ts [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    toastId = toast.error(content, options);
+  };
+
+  const connect = async (networkConfigUrl: string) => {
+    const { isWalletConnectionInProgress, chainConnection, importContext } =
+      appStore.getState();
+
+    if (isWalletConnectionInProgress || chainConnection) return;
+
+    let connection;
```
[the remark this answers is not in the document]
Comment (samsiegart) [reply src/service/wallet.ts]: Oh right, yea done

File: src/store/app.ts (hunk 1 above)
Comment (samsiegart) [reply src/store/app.ts]: Added to config.ts

File: src/vite-env.d.ts (hunk 2 above)
Comment (samsiegart) [reply src/vite-env.d.ts:15]: Added a comment

File: src/components/OfferSignerBridge.tsx [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+      { autoClose: 5000 },
+    );
+  };
+
+  const onBridgeReady = (ev: BridgeReadyMessage) => {
+    const {
+      detail: { isDappApproved, requestDappConnection, addOffer },
+    } = ev;
+    setOfferSigner({ addOffer, isDappApproved });
+    if (!isDappApproved) {
```
Comment (turadg) [inline src/components/OfferSignerBridge.tsx:104]: Nit: unnecessary inversion

File: src/service/wallet.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (turadg) [reply src/service/wallet.ts:18]: There should be a comment here so someone reading this isn't perplexed or tempted to delete.

Not blocking but consider leaving out this unused logic.

**Review by turadg — APPROVED** [review]

File: src/components/OfferSignerBridge.tsx (hunk 10 above)
Comment (samsiegart) [reply src/components/OfferSignerBridge.tsx]: Fixed

**Pull request merged**